battleship.py

```diff
@@ -106,17 +113,74 @@ def print_battle_board(self):
+        usr_input = str(usr_input).upper()
+        is_valid_bomb_target = False
+        error_message = 'ERROR: Please Enter a valid row(A-H) and column(0-7). Example: A5'
+        while is_valid_bomb_target is False:
+            ## checks if the user_input if the length of user input is 2 or not
+            if(len(usr_input) != 2):
+                self.display_message(error_message)
+                self.handle_user_input()
+                break
+            row = usr_input[0]
+            col = usr_input[1]
+            ## checks if the row is an alphabet and column is a number
+            if(not row.isalpha() or not col.isnumeric()):
+                self.display_message(error_message)
+                self.handle_user_input()
+                break
+            row = self.y_axis.find(row)
+            ##checks if the user entered row is outside of the battleboard
+            if not (-1 < row < self.battle_board_size):
+                self.display_message(error_message)
+                self.handle_user_input()
+                break
+            col = int(col)
+            ##checks if the user entered column is outside of the battleboard
+            if not (-1 < col < self.battle_board_size):
+                self.display_message(error_message)
+                self.handle_user_input()
+                break
+            if self.battle_board[row][col] == "#" or self.battle_board[row][col] == "X":
+                self.display_message("You have already shot a bomb here, pick somewhere else")
+                self.handle_user_input()
+                break
+            if self.battle_board[row][col] == ".":
+                is_valid_bomb_target = True
+                self.bomb_count = self.bomb_count - 1
+        return row,col
+
 
     ## updates the 8x8 board and the bomb count after the bomb is hit
     def update_battle_board_and_bomb_count(self, usr_input):
-        self.bomb_count = self.bomb_count - 1
-        print('@@@@@@@',usr_input)
+        row,col = self.check_bomb_placement_validity(usr_input)
+
+        #checks if the row,col value matches the key is ships dictionary
+        if (str(row)+str(col)) in self.ships:
+            self.display_message("It's a hit!! A battleship was hit by your bomb.")
+            self.ship_hit_counter += 1
+            self.battle_board[row][col] = "X"
+            if self.check_for_ship_destroyed(row, col):
+                self.display_message("Hell Yeah!! A ship was completely destroyed!")
+                self.num_of_ships_destroyed += 1
+        else:
+            self.display_message("It's a miss!! No battleship was hit.")
+            self.battle_board[row][col] = "#"
+        
+    ## TODO checks if a ship is completely destroyed
+    def check_for_ship_destroyed(self, row, col):
+        """If all parts of a shit have been shot it is destroyed we increment the num_of_ships_destroyed counter by 1"""
+        print("Hello I am a TODO")
```

Comment:
Minor item: I would prefer to see a 
```py
raise Execption("Not implemented")
```
Here instead of the print but I think this is fine. At the current rate this will be implemented soon enough

Reply:
Haha sorry man forgot to remove it...i was checking that for myself!!